src/main/java/com/artipie/SliceFromConfig.java

```diff
@@ -139,7 +140,18 @@ static Slice build(final Settings settings, final Authentication auth,
         final Permissions permissions = new LoggingPermissions(
             cfg.permissions().orElse(Permissions.FREE)
         );
-        final Pattern prefix = new PathPattern(settings).pattern();
+        final Pattern prefix;
+        if (cfg.port().isEmpty()) {
+            prefix = new PathPattern(settings).pattern();
+        } else {
+            // A dirty-hacked patter for not trimming anything.
+            // Since TrimPathSlice keeps only the group number 1
+            // we don't want anything to be removed from the path
+            // in case a repo is served on a different port;
+            // that is why we need an empty group in the beginning
+            // of the pattern.
+            prefix = Pattern.compile("()(/.*)?");
+        }
```

Comment:
@Sammers21 there is no test coverage for this code

Reply:
@olegmoz, how should i test it?